packages/tools/src/utilities/math/polyline/getLineSegmentsIntersection.ts

```diff
@@ -30,7 +30,10 @@ export default function getLineSegmentsIntersection(
   const denominator = diffQ2P2[1] * diffQ1P1[0] - diffQ2P2[0] * diffQ1P1[1];
 
   if (denominator == 0) {
-    return;
+    return [
```

Comment:
@lscoder what do you think

Reply:
Returning only a mid point does not look right because the intersection would be an infinity number of points (range). In which case this happens and why the midpoint would solve the problem?

Reply:
Also the function is returning wrong values:
- ([0, 0], [4, 4], [5, 5], [10, 10]): 4.75 (the lines do not even intersect)
- ([0, 0], [4, 4], [3, 3], [20, 20]): 6.75 (The intersection goes from [3, 3] to [4, 4])

Reply:
That was pre-existing - the lines never needed to intersect to return a value.  Will write a unit test and add a flag to have true intersection or midpoint etc.  Returning undefined isn't handled by either use case, and I don't know enough about those use cases to fix them.

Reply:
Issue resolved with the changes I pushed